### DISCUSSION TITLE
CustomTkinter Support

### DISCUSSION THREAD

**Discussion by Govine-Jarrett** [alejandroautalan/pygubu-designer#156] [Q&A] [answered]
How can I add support for [CustomTkinter](https://github.com/TomSchimansky/CustomTkinter) ?

**Comment by alejandroautalan** [comment]
Hello @Govine-Jarrett, thanks for trying pygubu.

I have already been reviewing CustomTkinter, to make a plugin. Its incorporation will require changes in the designer itself so it will take me a while to finalize it.
If you look [here](https://github.com/alejandroautalan/pygubu/tree/master/src/pygubu/plugins/tkintermapview), you will see that I have already made a plugin for TkinterMapView but, it is not advertised since it also requires some changes in the designer .
In one or two weeks I will upload my progress, so you can help me to test and fix issues if you like.

Regards
Alejandro A.

**Reply by Govine-Jarrett** [reply]
I am actively using the designer  since it was introduced to me by @jrezai via his YouTube tutorials, I have to say that this is a really powerful tool. 

It would be an honor to help you test out any plugins that you may create for the designer.

**Comment by alejandroautalan** [comment, marked as answer]
Hello @Govine-Jarrett and all interested in customtkinter support.
I merged the changes in the master branch.
You can try and install in a test environment with the following commands:

```bash
python3 -m venv test-venv
source test-venv/bin/activate
pip install customtkinter
pip install git+https://github.com/alejandroautalan/pygubu
pip install git+https://github.com/alejandroautalan/pygubu-designer
```
Any comments or suggestions please share it here.
I will continue testing to prepare a final release.

Here is a preview

![Captura de pantalla_2022-11-04_00-52-23](https://user-images.githubusercontent.com/3482471/199883051-0c8ecac1-332f-4e48-beab-b8da42d5a4b1.png)

Regards!
Alejandro A.

**Reply by Govine-Jarrett** [reply]
@alejandroautalan thank you, I will keep you posted

**Comment by iradraconis** [comment]
in which version will this be available? I habe installed 0.31.3 of pygubu-designer and 0.26.2 of pygubu via pip and there is no customtkinter support.

**Reply by alejandroautalan** [reply]
Hello @iradraconis,
I uploaded a new version. Please report any issue you found.

    pip install --upgrade pygubu-designer

Regards!
Alejandro A.
